server/game/cards/01_SOR/events/SneakAttack.ts

```diff
@@ -0,0 +1,39 @@
+import { EventCard } from '../../../core/card/EventCard';
+import AbilityHelper from '../../../AbilityHelper';
+import { CardType, PhaseName, ZoneName } from '../../../core/Constants';
+import { CostAdjustType } from '../../../core/cost/CostAdjuster';
+
+export default class SneakAttack extends EventCard {
+    protected override getImplementationId() {
+        return {
+            id: '3426168686',
+            internalName: 'sneak-attack'
+        };
+    }
+
+    public override setupCardAbilities() {
+        this.setEventAbility({
+            title: 'Play a unit from your hand. It costs 3 less and enters play ready. At the start of the regroup phase, defeat it.',
+            immediateEffect: AbilityHelper.immediateEffects.selectCard({
+                activePromptTitle: 'Play a unit, it costs 3 less and enters play ready. At the start of the regroup phase, defeat it.',
```

Comment:
Does the prompt not show up correctly without this? selectCard might be missing some features

Reply:
@AMMayberry1 Do you confirm we should use `selectCard` here ? I tried with `targetResolver` as you suggested for holdup blaster but I found it a bit less convenient because it allows to select a card even if we do not have enough resources to play it.
Is it something missing from targetResolver (I looked at alliance dispatcher or Han solo set 2) and they both do not filter on the cards we can play after the discount, or is it me not knowing yet how to achieve it?

Reply:
Yeah that must be some kind of bug, if a target is not legal for an effect then it shouldn't up in the selection list for it. I'll take a look at this when I get a chance.

Reply:
Ah okay, I figured out what's going on. It's a little bit tricky (and `selectCard()` isn't behaving correctly for some reason).

Basically, since swu ability resolution is "do as many effects as possible" (i.e. it's okay if some effects don't happen), the `sequential()` and `simultaneous()` operators will allow a target to be selected if it's legal for at least one of the effects in their list. In this case, any target in hand is technically legal for the DelayedEffect just because that doesn't have any hard restrictions on what cards are legal for it to target.

So, this is like some other cases we've seen where there's really an implicit "if you do" in the card. Even though it doesn't technically say it, since the latter effect (the delayed effect) is meaningless if the first effect didn't happen, logically speaking it's the same as "if you do." Some cards like Greedo have the same issue and we solved them the same way - in Greedo's case, the damage can't resolve if no discard happened since there's no card to check to see if it's a unit.

So in this case let's do the play action as a normal effect inside of a targetResolver (no `sequential`) and then do the delayed effect as an `ifYouDo`. Feel free to ping me in chat if that doesn't make sense.